starcheck/src/starcheck.pl

```diff
@@ -108,6 +109,45 @@
     return state
 
 
+def get_now_time():
+    """
+    Return the current time in YYYY:DOY format
+    """
+    return DateTime().date
+
+
+def run_start_time(run_start, backstop_start):
+    """
+    Determine a reasonable reference run start time based on the supplied
+    run start time and the time of the first backstop command.  This
+    code uses a small hack so that a negative number is interpreted
+    as the desired "days back" from the backstop start time.  All other
+    Chandra.Time compatible formats for run start are used as absolute
+    times (which will then be passed to the thermal model code as the
+    time before which telemetry should be found for an initial state.
+
+    :param run_start: supplied run start time in a Chandra.Time format,
+                      negative numbers interpreted as "days back" relative
+                      to first backstop command
+    :param backstop_start: time of first backstop command
+    :returns: YYYY:DOY string of reference run start time
+    """
+
+    run_start = de_bytestr(run_start)
+    ref_time = DateTime(run_start)
+    # For the special case where run_start casts as a float
+    # check to see if it is negative and if so, override the reference
+    # time with a time run_start days back from backstop start
+    try:
+        run_start = float(run_start)
+        if run_start < 0:
+            backstop_datetime = DateTime(de_bytestr(backstop_start))
```

Comment:
The `de_bytestr` processing should always be a separate line at the top of the function.  Separate the function input wrangling from the actual processing.  Then this just becomes one line:
```
ref_time = DateTime(backstop_start) + run_start  # Already a DateTime, no recast required
```